gtsfm/bundle/bundle_adjustment_hilti.py

```diff
@@ -0,0 +1,189 @@
+        hilti_loader = HiltiLoader(
+            base_folder=str(HILTI_TEST_DATA_PATH),
+            max_frame_lookahead=1,
+            max_length=None,
+        )
+
+        self._cam_T_imu = hilti_loader.get_camTimu()
+
+    def __get_rig_idx(self, camera_idx: int) -> int:
+        return camera_idx // 5
+
+    def __get_camera_type(self, camera_idx: int) -> int:
+        return camera_idx % 5
+
+    def _between_factors(
+        self, relative_pose_priors: Dict[Tuple[int, int], Optional[PosePrior]], cameras_to_model: List[int]
+    ) -> NonlinearFactorGraph:
+        graph = NonlinearFactorGraph()
+
+        # add a between factor for each camera with its IMU
+        for i in cameras_to_model:
+            camera_type: int = self.__get_camera_type(i)
+            rig_idx: int = self.__get_rig_idx(i)
+            graph.push_back(
+                BetweenFactorPose3(
+                    X(i),
+                    B(rig_idx),
+                    self._cam_T_imu[camera_type],
+                    gtsam.noiseModel.Isotropic.Sigma(CAM_POSE3_DOF, CAM_IMU_POSE_PRIOR_SIGMA),
+                )
+            )
+
+        # translate the relative pose priors between cams to IMUs, and add if not already present
+        imu_relative_pose_priors: Dict[Tuple[int, int], BetweenFactorPose3] = {}
+        for (i1, i2), i2Ti1_prior in relative_pose_priors.items():
+            if i2Ti1_prior is None or i1 not in cameras_to_model or i2 not in cameras_to_model:
+                continue
+
+            b1: int = self.__get_rig_idx(i1)
+            b2: int = self.__get_rig_idx(i2)
+
+            if b1 == b2:
+                # already captured in cam-IMU between factor.
+                continue
+
+            if (b1, b2) in imu_relative_pose_priors:
+                continue
+
+            if i2Ti1_prior.type == PosePriorType.HARD_CONSTRAINT:
+                raise ValueError("Encountered hard pose constraint in inter-rig camera pairs")
+
+            i2Ti1 = i2Ti1_prior.value
+            i1Tb1 = self._cam_T_imu[self.__get_camera_type(i1)]
+            i2Tb2 = self._cam_T_imu[self.__get_camera_type(i2)]
+
+            b2Tb1 = i2Tb2.inverse() * i2Ti1 * i1Tb1
+            imu_relative_pose_priors[(b1, b2)] = BetweenFactorPose3(
+                B(b2),
+                B(b1),
+                b2Tb1,
+                gtsam.noiseModel.Diagonal.Sigmas(i2Ti1_prior.covariance),
+            )
+
+        logger.info("Adding %d between factors for IMUs", len(imu_relative_pose_priors))
+
+        for factor in imu_relative_pose_priors.values():
+            graph.push_back(factor)
+
+        return graph
+
+    def _pose_priors(
+        self,
+        absolute_pose_priors: List[Optional[PosePrior]],
+        initial_data: GtsfmData,
+        camera_for_origin: gtsfm_types.CAMERA_TYPE,
+    ) -> NonlinearFactorGraph:
+        graph = NonlinearFactorGraph()
+
+        # TODO(Ayush): start using absolute prior factors.
```

Comment:
wait, so what *is* this function doing? docstring missing

Reply:
Just adding the prior on 1 IMU to define the origin of the world frame.